src/Internal/Payment/State/AuthenticationRequiredState.php

```diff
@@ -0,0 +1,75 @@
+<?php
+/**
+ * Class AuthenticationRequiredState
+ *
+ * @package WooCommerce\Payments
+ */
+
+namespace WCPay\Internal\Payment\State;
+
+use WCPay\Internal\Service\CheckoutEncryptionService;
+
+/**
+ * The state, which indicates that the payment processing has been completed.
+ */
+class AuthenticationRequiredState extends AbstractPaymentState {
+	/**
+	 * Checkout Encryption service
+	 *
+	 * @var CheckoutEncryptionService
+	 */
+	private $checkout_encryption_service;
+
+
+	/**
+	 * Class constructor, only meant for storing dependencies.
+	 *
+	 * @param StateFactory              $state_factory                            Factory for payment states.
+	 * @param CheckoutEncryptionService $checkout_encryption_service Service for encrypting checkout data.
+	 */
+	public function __construct(
+		StateFactory $state_factory,
+		CheckoutEncryptionService $checkout_encryption_service
+	) {
+		parent::__construct( $state_factory );
+
+		$this->checkout_encryption_service = $checkout_encryption_service;
+	}
+	/**
+	 * Get response that will be sent to the client.
+	 */
+	public function get_response(): array {
```

Comment:
This should probably not live in the state, as the redirect (especially 3DS) is very gateway and JS-specific. On the other hand, it is just a string, so maybe it makes sense. I'd like @marcinbot 's opinion here as well.

Reply:
Good point. Addressed here: https://github.com/Automattic/woocommerce-payments/pull/7471/commits/044cdf8890f23a366b897fef839b1c28c29bf06a